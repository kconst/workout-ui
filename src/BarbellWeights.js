import React, { Component, PropTypes } from 'react'
import classnames from 'classnames'

class BarbellWeights extends Component {
    constructor(props, context) {
        super(props, context);
        this.state = {
            weight: 0
        }
    }

    componentWillMount() {
        this.setState({
           weight: this.props.weight 
        });
    }

    calcWeights(weight, isImperial) { 
        let weights = [],
            barbellWeight,
            available;

        if (isImperial) {
            available = [2.5, 5, 10, 25, 45].reverse();
            
            barbellWeight = 45;
        } else {
            available = [2.5, 5, 10, 20, 25].reverse();
            
            barbellWeight = 20;
        }

        weight = weight - barbellWeight;

        if (!weight || weight < 5) {
            console.warn('Invalid input.');
            
            return [];
        }

        available.forEach((e) => {
            while(weight && (weight / 2)  >= e) {
                weight = weight - (e * 2);
                weights.push(e);
            }
        });

        return weights
    }

    render() {
        // const { todo, completeTodo, deleteTodo } = this.props;

        let element,
            plates = this.calcWeights(this.props.weight);

        element = (
            <ul className="plate-list">
                {plates.map((plate, i) =>
                    <li key={i} className={ 'plate-' + (plate + '').replace('.', 'p') }>{ plate }</li>
                )}
            </ul>
        );

        return (
            element
        );
    }
}

BarbellWeights.propTypes = {
    
};

export default BarbellWeights
