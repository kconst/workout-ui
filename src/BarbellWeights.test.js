import React from 'react';
import BarbellWeights from './BarbellWeights.js';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

// fixes enzyme's issue with jest
jest.mock('react/lib/ReactDefaultInjection');

it('renders correctly in default state', () => {
  const component = renderer.create(<BarbellWeights useImperial={ false } weight={ 100 }/>);
  
  let weights = component.toJSON();
  expect(weights).toMatchSnapshot();
  
  
  expect(renderer.create(<BarbellWeights useImperial={ true } weight={ 225 }/>).toJSON()).toMatchSnapshot();
});

/*it ('should have the correct total for metric weights', () => {
  const component = shallow(<PlatesSelect useImperial={ false }/>);
  
  component.find('.plate-15').simulate('click');
  
  expect(component.find('h3').text()).toEqual('50');
});*/