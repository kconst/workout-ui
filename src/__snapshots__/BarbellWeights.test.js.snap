exports[`test renders correctly in default state 1`] = `
<ul
  className="plate-list">
  <li
    className="plate-25">
    25
  </li>
  <li
    className="plate-10">
    10
  </li>
  <li
    className="plate-5">
    5
  </li>
</ul>
`;

exports[`test renders correctly in default state 2`] = `
<ul
  className="plate-list">
  <li
    className="plate-25">
    25
  </li>
  <li
    className="plate-25">
    25
  </li>
  <li
    className="plate-25">
    25
  </li>
  <li
    className="plate-25">
    25
  </li>
  <li
    className="plate-2p5">
    2.5
  </li>
</ul>
`;
